synapse/rest/client/v2_alpha/account_data.py

```diff
@@ -47,6 +49,11 @@ def on_PUT(self, request, user_id, account_data_type):
 
         body = parse_json_object_from_request(request)
 
+        if account_data_type == "im.vector.hide_profile":
+            user = UserID.from_string(user_id)
+            hide_profile = body.get('hide_profile')
+            yield self._profile_handler.set_active(user, hide_profile, True)
```

Comment:
Do you mean to set the value of `active` to hide_profile and the value of `hide` to True ? I think this wants to be the other way around?

Reply:
sorry, this is confusing. the order of the parameters is correct: the value of `hide` should indeed be `True`, as it's telling set_active to edit the hiddenness of the profile rather than its activity.

however, I completely thinkoed by making `{hide_profile:  false}` mean "hide the profile" rather than "show the profile". So i've fixed that now.